examples/django/oauth_app/slack_datastores.py

```diff
@@ -33,13 +33,50 @@ def save(self, installation: Installation):
         i = installation.to_dict()
         if is_naive(i["installed_at"]):
             i["installed_at"] = make_aware(i["installed_at"])
+        if "bot_token_expires_at" in i and is_naive(i["bot_token_expires_at"]):
+            i["bot_token_expires_at"] = make_aware(i["bot_token_expires_at"])
+        if "user_token_expires_at" in i and is_naive(i["user_token_expires_at"]):
+            i["user_token_expires_at"] = make_aware(i["user_token_expires_at"])
         i["client_id"] = self.client_id
-        SlackInstallation(**i).save()
-        b = installation.to_bot().to_dict()
+        row_to_update = (
```

Comment:
Changed the save method for token rotation. When refreshing tokens happens, we update the existing database row instead of inserting a new one.